gcc/config/riscv/riscv.c

```diff
@@ -3609,12 +3630,7 @@ riscv_hard_regno_mode_ok_p (unsigned int regno, enum machine_mode mode)
     }
 
   if (FP_REG_P (regno))
-    {
-      if (mclass == MODE_FLOAT
-	  || mclass == MODE_COMPLEX_FLOAT
-	  || mclass == MODE_VECTOR_FLOAT)
-	return size <= UNITS_PER_HWFPVALUE;
-    }
+    return true;
```

Comment:
btw, it's also make gpr<-> fpr spilling is possible if register allocator think it's worth (depend on cost model).